Snowflake.Data/Core/SFStatement.cs

```diff
@@ -180,6 +183,10 @@ private void CleanUpCancellationTokenSources()
 
         private SFBaseResultSet BuildResultSet(QueryExecResponse response, CancellationToken cancellationToken)
         {
+            if (response.data.queryId != null)

```

Comment:
Response.data should be there but I would still check if it's not null

Reply:
done